### PR TITLE
reduce logs on every received message

### PR DESCRIPTION
**Pull request by gadcl**
only log on view transition (from old to new): logs all (members-> messages list) of old view

### CODE REVIEW

File: services/storage/in_memory_storage.go [hunk 1]
```diff
@@ -358,12 +359,97 @@ func (storage *InMemoryStorage) resetCommitStorage(blockHeight primitives.BlockH
...
 	return views
 }
+
 func (storage *InMemoryStorage) resetViewChangeStorage(blockHeight primitives.BlockHeight) map[primitives.View]map[MemberIdStr]*interfaces.ViewChangeMessage {
 	views := make(map[primitives.View]map[MemberIdStr]*interfaces.ViewChangeMessage)
 	storage.viewChangeStorage[blockHeight] = views
 	return views
 }
 
+func (storage *InMemoryStorage) GetMessagesLogs(blockHeight primitives.BlockHeight, view primitives.View) []interfaces.MemberMessagesLog {
```
Comment (ronnno) [inline services/storage/in_memory_storage.go:369]: its a little odd that in_memory_storage formats your logs messages for you.. I realize you use non public members here but maybe the better fix is to make getter functions to all the data you need to format the log. this way you don't need to define a new struct "MessagesLog"..

why not do all this logic in some logger object? or at least some helper method of `logViewMessages`.

At least, rename this to "AsMessageLog" or something similar to indicate this changes the representation of the state.

File: services/storage/in_memory_storage.go [hunk 2]
```diff
@@ -358,12 +359,97 @@ func (storage *InMemoryStorage) resetCommitStorage(blockHeight primitives.BlockH
...
+	memberMessages := make(map[MemberIdStr][]*protocol.BlockRef)
+	// get preprepare
+	if msg, exists := storage.GetPreprepareFromView(blockHeight, view); exists {
+		memberIdStr := MemberIdStr(msg.Content().Sender().MemberId())
+		msgs, _ := memberMessages[memberIdStr]
```
Comment (ronnno) [inline services/storage/in_memory_storage.go:374]: a little verbose - you don't need the variable `msgs` if you change the append statement to work on `memberMessages[memberIdStr]`

File: services/storage/in_memory_storage.go [hunk 3]
```diff
@@ -358,12 +359,97 @@ func (storage *InMemoryStorage) resetCommitStorage(blockHeight primitives.BlockH
...
 func (storage *InMemoryStorage) resetViewChangeStorage(blockHeight primitives.BlockHeight) map[primitives.View]map[MemberIdStr]*interfaces.ViewChangeMessage {
 	views := make(map[primitives.View]map[MemberIdStr]*interfaces.ViewChangeMessage)
 	storage.viewChangeStorage[blockHeight] = views
 	return views
 }
 
+func (storage *InMemoryStorage) GetMessagesLogs(blockHeight primitives.BlockHeight, view primitives.View) []interfaces.MemberMessagesLog {
+	memberMessages := make(map[MemberIdStr][]*protocol.BlockRef)
+	// get preprepare
+	if msg, exists := storage.GetPreprepareFromView(blockHeight, view); exists {
```
Comment (ronnno) [inline services/storage/in_memory_storage.go:372]: do you need this `if` here? a for loop will never run if there are not elements

File: services/storage/in_memory_storage.go (hunk 1 above)
Comment (ronnno) [inline services/storage/in_memory_storage.go:369]: if you have the timestamp each message was received, or at least preserve the order in which they came in, that could be a nice addition to the data we output to the logs.

perhaps add a //TODO to add this?

**Review by ronnno — APPROVED** [review]
Hi,

Looks good.
- I did not check that you dump the logs in the right places. i wonder how we can gain confidence in that it's done always. maybe @vistra has an idea?
- there is a lot of code which is running when we try to dump the messages to the log file. i'm a little concerned about bugs in this code so maybe wrap `logViewMessages` with a `defer func(){if e:= recover; e != nil{ // log e...}}`

**Review by noambergIL — APPROVED** [review]
I don't know what's my opinion on one super big log line ... are we sure it won't break elks in the future ?@ron

File: services/storage/in_memory_storage.go (hunk 1 above)
Comment (gadcl) [reply services/storage/in_memory_storage.go:369]: I'm not sure it adds any reliable information regarding other nodes that we can rely on - kinda mixing network and in-node processing timing issues, but I added a note to reconsider

Comment (gadcl) [reply services/storage/in_memory_storage.go:369]: refactored

**Pull request merged**